src/audio_device.h

```diff
@@ -55,42 +56,42 @@ typedef enum DeviceError {
 typedef void (*DataHandleCallback)(const int16_t *, uint32_t size, void *data);
 
 
-#ifdef AUDIO
-DeviceError init_devices(ToxAV *av);
-#else
 DeviceError init_devices(void);
-#endif /* AUDIO */
 
-void get_devices_names(void);
+void get_al_device_names(void);
 DeviceError terminate_devices(void);
 
-/* Callback handles ready data from INPUT device */
-DeviceError register_device_callback(int32_t friend_number, uint32_t device_idx, DataHandleCallback callback,
-                                     void *data, bool enable_VAD);
-void *get_device_callback_data(uint32_t device_idx);
-
 /* toggle device mute */
 DeviceError device_mute(DeviceType type, uint32_t device_idx);
 
-#ifdef AUDIO
-DeviceError device_set_VAD_treshold(uint32_t device_idx, float value);
-#endif
+bool device_is_muted(DeviceType type, uint32_t device_idx);
+
+DeviceError device_set_VAD_threshold(uint32_t device_idx, float value);
+
+float device_get_VAD_threshold(uint32_t device_idx);
+
+DeviceError set_source_position(uint32_t device_idx, float x, float y, float z);
+
+DeviceError set_al_device(DeviceType type, int32_t selection);
 
-DeviceError set_primary_device(DeviceType type, int32_t selection);
-DeviceError open_primary_device(DeviceType type, uint32_t *device_idx, uint32_t sample_rate, uint32_t frame_duration,
-                                uint8_t channels);
 /* Start device */
-DeviceError open_device(DeviceType type, int32_t selection, uint32_t *device_idx, uint32_t sample_rate,
-                        uint32_t frame_duration, uint8_t channels);
+DeviceError open_input_device(uint32_t *device_idx,
+                              DataHandleCallback cb, void *cb_data, bool enable_VAD,
+                              uint32_t sample_rate, uint32_t frame_duration, uint8_t channels);
+DeviceError open_output_device(uint32_t *device_idx,
+                               uint32_t sample_rate, uint32_t frame_duration, uint8_t channels);
+
 /* Stop device */
 DeviceError close_device(DeviceType type, uint32_t device_idx);
 
-/* Write data to device */
+/* Write data to output device */
 DeviceError write_out(uint32_t device_idx, const int16_t *data, uint32_t length, uint8_t channels,
                       uint32_t sample_rate);
 
-void print_devices(ToxWindow *self, DeviceType type);
-void get_primary_device_name(DeviceType type, char *buf, int size);
+/* return current input volume as float in range 0.0-100.0 */
+float get_input_volume();
```

Comment:
`(void)`

If you don't put anything in the (), these become variadic functions, so we have no guards against accidentally passing parameters.